ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Wallets.hs

```diff
@@ -175,3 +200,25 @@ updateHdWalletAssurance
     -> IO (Either HD.UnknownHdRoot HdRoot)
 updateHdWalletAssurance pw hdRootId assuranceLevel =
     update' (pw ^. wallets) (UpdateHdWalletAssurance hdRootId assuranceLevel)
+
+{-------------------------------------------------------------------------------
+  Automatic creation of Wallet Account and Address
+-------------------------------------------------------------------------------}
+
+addAddress 
+    :: PassiveWallet
+    -> WalletId
+    -> PassPhrase
+    -> HdRoot
+    -> IO (Either CreateWalletError HdRoot)
+addAddress pw walletId passphrase hdRoot = do 
+    accountRes <- createAccount (AccountName "un-named") walletId pw
+    case accountRes of 
+        Left err      -> return . Left $ CreateAccountFailed err
+        Right account -> do
+            let accountId = AccountIdHdSeq $ account ^. HD.hdAccountId
+            addressRes <- createAddress passphrase accountId HdChainInternal pw
+            case addressRes of
+                Left err -> return . Left $ CreateAddressFailed err
+                Right _  -> return (Right hdRoot)
```

Comment:
I'm also thinking that if creation of a new account/address fails, then this should be treated as a programmer error (this should never happen), and so all of these `Create{Account,Address}Failed` could be removed and the corresponding errors thrown with `impureThrow` according to the [exception guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md#potentially-impure-code-programmer-mistakes). However, this looks like a border case, and I'm OK with keeping all of these constructors too.